.github/workflows/promote.yaml

```
@@ -0,0 +1,36 @@
+name: Promote snap to other tracks and channels
+
+on:
+  workflow_dispatch:
+    inputs:
+      destination-channel:
+        description: 'Destination Channel, e.g. latest/candidate'
+        required: true
+      origin-channel:
+        description: 'Origin Channel, e.g. latest/edge'
+        required: true
+
+jobs:
+  promote-snap:
+    name: Promote snap
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install yq
+        run: sudo snap install yq
+      - name: Snapcraft install
+        run: sudo snap install --classic snapcraft
+      - name: Get snap name
+        id: snap
+        run: echo "name=$(yq '.name' snap/snapcraft.yaml)" >> $GITHUB_OUTPUT
```

given how simple the parsing you want to do of `snapcraft.yaml` is, I'd suggest not including an extra dependency (especially after the xz mess...)

```suggestion
      - name: Snapcraft install
        run: sudo snap install --classic snapcraft
      - name: Get snap name
        id: snap
        run:  grep -Po '(?<=^name:)\s*\K\w+' snap/snapcraft.yaml >> $GITHUB_OUTPUT
```